Add ACL Recipes section to HowTo

fdevans opened the pull request:
Initial push of the ACL Recipes section.

fdevans committed:
Establish ACL Recipes Start

fdevans committed:
Updated to make nav listing dynamic

fdevans committed:
fix sort lowest to highest

fdevans committed:
More cleanup

fdevans committed:
Cross reference existing cotent.

fdevans committed:
bold

wayfaringson committed:
A few typo fixes and four new acl recipe snippets

fdevans commented on docs/learning/howto/acls/group-apikey.md:
```diff
@@ -0,0 +1,32 @@
+
+# Group/API Key Create User API Ke
+
+## Use Case Description
```
@wayfaringson - I think we should provide more background here.  Are users not allowed to create API keys by default?

wayfaringson replied:
Because it is a system task, i don't believe they can do so without this (unless they are admin).

fdevans replied:
this is a good add then

fdevans commented on docs/learning/howto/acls/group-jobname.md:
```diff
@@ -0,0 +1,50 @@
+order: 500
+---
+
+# Group/Project/JobName Execute Only
```
We should also allow read access to the project Key Storage, or note that if the jobs uses Key Storage entries it will need to use something to provide those rights.

fdevans committed:
minor touch ups

fdevans committed:
No need for project, improve group name

fdevans committed:
Update group-project-exec.md

fdevans committed:
Update group-apikey.md

wayfaringson committed:
Updates to API key and job group/name

Added context for the API key rule and a clause for access to project keys to the two job execution policies

wayfaringson committed:
Update group-manage-runner.md

wayfaringson committed:
Update group-jobname.md

wayfaringson reviewed (approved):
I tested all of the ACL snippets and caught two things that needed correcting which are now reflected here.  As far as I am concerned, this could be merged at this point.
